internal/tools/update-api-version/main.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+var logger hclog.Logger
+
+func main() {
+	logger = hclog.New(hclog.DefaultOptions)
+	if os.Getenv("DEBUG") != "" {
+		logger.SetLevel(hclog.Debug)
+	}
+
+	f := flag.NewFlagSet("update-api-version", flag.ExitOnError)
+	serviceName := f.String("service", "", "-service=compute")
+	oldApiVersion := f.String("old-api-version", "", "-old-api-version=2019-01-01")
+	newApiVersion := f.String("new-api-version", "", "-new-api-version=2023-06-01")
+	_ = f.Parse(os.Args[1:])
+
+	if err := f.Parse(os.Args); err != nil {
+		log.Fatalf("parsing arguments: %+v", err)
+	}
```

Comment:
Should we also check that we actually have 2 args here? (e.g. `f.NArgs() == 2` or check non-empty strings?

Reply:
Good shout - adding